src/sentry/snuba/metrics/utils.py

```diff
@@ -145,6 +145,10 @@ class MetricDoesNotExistException(Exception):
     ...
 
 
+class MetricDoesNotExistInIndexer(Exception):
```

Comment:
Note that we have a utility with similar purpose in `sentry.sentry_metrics.utils`, but I don't think we should reuse their exceptions.